sw/host/tests/rom/e2e_bootstrap_rma/src/main.rs

```diff
@@ -18,359 +23,261 @@ use opentitanlib::dif::lc_ctrl::{
+        .context("failed to interact with console")?;
+
+    match result {
+        ExitStatus::ExitSuccess => Ok(()),
+        result => bail!("FAIL: {result:?}"),
+    }
 }
 
-/// Orchestrates iteration 1 of the rom_bootstrap_rma testpoint. This never
-/// issues the life cycle RMA command. It verifies that the ROM times out on
-/// spin cycles, automatically resets the device, and logs the expected reset
-/// reasons.
-fn test_no_rma_command(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
-    let uart = transport.uart("console")?;
-    reset(
-        transport,
-        &["RMA_BOOTSTRAP"],
-        opts.init.bootstrap.options.reset_delay,
+/// Iteration 2 of the `rom_bootstrap_rma` testpoint.
+///
+/// Issues the lifecycle RMA command and ensures the RMA transition is completed
+/// successfully.
+fn test_rma_command(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    let rma_bootstrap_strapping = transport.pin_strapping("RMA_BOOTSTRAP")?;
+    let tap_strapping = transport.pin_strapping("PINMUX_TAP_LC")?;
+
+    // Need to reset with `RMA_BOOTSTRAP` set.
+    rma_bootstrap_strapping
+        .apply()
+        .context("failed to apply RMA_BOOTSTRAP strapping")?;
+
+    // We keep `PINMUX_TAP_LC` set so that resets due to transition don't
+    // disconnect us from the LC TAP.
+    tap_strapping
+        .apply()
+        .context("failed to apply PINMUX_TAP_LC strapping")?;
+
+    transport
+        .reset_target(opts.init.bootstrap.options.reset_delay, true)
+        .context("failed to reset target")?;
+
+    // Remove the RMA_BOOTSTRAP strapping to bring the device out of the RMA spin loop.
+    rma_bootstrap_strapping
+        .remove()
+        .context("failed to remove strapping RMA_BOOTSTRAP")?;
+
+    // NOTE: without this sleep, the transition starts but isn't successful.
+    // The sleep must happen anywhere before the `START` `TRANSITION_CMD` is given.
+    // Not sure why at the moment, seems like the chip isn't ready even though the
+    // `STATUS.READY` bit is checked.
+    std::thread::sleep(Duration::from_secs(1));
+
+    log::info!("Connecting to JTAG interface");
+
+    let jtag = transport.jtag(&opts.jtag).context("failed to get JTAG")?;
+    jtag.connect(JtagTap::LcTap)
+        .context("failed to connect to JTAG")?;
+
+    // Wait for the lifecycle controller to enter the `READY` state from
+    // which it accepts transition commands.
+    lc_transition::wait_for_status(&jtag, Duration::from_secs(3), LcCtrlStatus::READY)
+        .context("failed to wait for lifecycle controller to be ready")?;
+
+    // Check we're in the `PROD` state with 5 transitions registered.
+    assert_eq!(
+        jtag.read_lc_ctrl_reg(&LcCtrlReg::LcState)?,
+        DifLcCtrlState::Prod.redundant_encoding()
+    );
+    assert_eq!(jtag.read_lc_ctrl_reg(&LcCtrlReg::LcTransitionCnt)?, 5);
+
+    // Attempt to claim the transition mutex.
+    jtag.write_lc_ctrl_reg(
```

Comment:
are you going to update the `trigger_lc_transition` testutil function to accept a token and use it in this PR? or are you going to do that in a follow up PR?

Reply:
@jwnrt and I discussed offline, this will be done in a follow up PR